parcels/particlefile/baseparticlefile.py

```diff
@@ -213,15 +221,20 @@ def _create_trajectory_records(self, coords):
             self.z.units = "m"
             self.z.positive = "down"
 
-        for vname in self.var_names:
+
+        for vname, dtype in zip(self.var_names, self.var_dtypes):
             if vname not in self._reserved_var_names():
-                setattr(self, vname, self.dataset.createVariable(vname, "f4", coords, fill_value=np.nan))
+                fill_value = np.nan if dtype[0] == 'f' else np.iinfo(np.dtype(dtype)).min
```

Comment:
I am not entirely confident about the `np.iinfo(np.dtype(dtype)).min` filling value. That would result for unsigned integer types in `0`, which can easily conflict with an actually useful value. I think `np.iinfo(np.dtype(dtype)).max` would be more resilient. Of course, it's a design decision - the convention doesn't give a firm limit on the fill-value.

Reply:
I think that's sensible - I just went with the minimum value since I have rarely (if ever) come across unsigned ints in a netcdf, but for versatility this seems like a good idea.